src/coord/timestamp.rs

```diff
@@ -206,6 +218,23 @@ fn byo_query_source(consumer: &mut ByoTimestampConsumer, max_increment_size: i64
     let mut msg_count = 0;
     match &mut consumer.connector {
         ByoTimestampConnector::Kafka(kafka_consumer) => {
+            let kconsumer = &kafka_consumer.consumer;
+            let topic = &consumer.source_name;
+            for p in get_kafka_partitions(kconsumer, &topic) {
+                match kconsumer.fetch_watermarks(&topic, p, Duration::from_secs(1)) {
+                    Ok((_low, high)) => CURRENT_KAFKA_PARTITION_OFFSET
```

Comment:
This concerns me a bit. We are adding another loop that sends N pings to Kafka every 10ms where N is the number of sources. I'm less concerned about the RT case because we are already doing it there, whereas we aren't in BYO world.

Not sure if it's a real issue, but adding 100 * n_sources QPS purely for the sake of generating metrics just intuitively seems off to me.

I'm spitballing here, but, maybe the right thing to do is just keep a timer since the last time we did this fetch, and only do it if we haven't done it in >1s or so?

Reply:
In the RT case we have to do it, but we don't read from Kafka. In the BYO case, we don't have to use any metadata information, but we read from Kafka.

Is there a reason why we couldn't have a separate "thread" that (with lower intervals), collects statistics?  In the same way as we have the update()  function which updates ts information. Would it make more sense to have a "collect_stats" function that does this?

Reply:
That is a good idea, on the other hand, I liked doing it inline in the RT case because we are already getting that data anyway. But if we put it on another thread and only did it once every 1s or something, it probably wouldn't generate enough load on Kafka to matter.

Reply:
ah yeah in my head this also only happened on an interval, but that doesn't make senseone of the major benefits of of BYO consistency is it can be faster than that, yeah?

Since we already have it in the RT case I'd rather not move it out to a different thread.

Reply:
I was getting started moving this out to a different thread but this does not need to be super accurate (most external aggregations will only be hitting this every 10-60 seconds anyway) so I don't think we need to update more than once per second per topic, so I think I'll just clamp this update to 1 update per topic per second unless someone has an objection. We can always make it configurable (later) if that is a problem.

Reply:
SGTM, it sounds like that’s the easiest thing to do that doesn’t cause too much extra load.